coalib/output/ConsoleInteraction.py

```diff
@@ -922,27 +933,58 @@ def print_bears(bears,
     :param show_params:      True if the parameters and their description
                              should be shown.
     :param console_printer:  Object to print messages on the console.
+    :param args:             Args passed to coala command.
     """
     if not bears:
         console_printer.print('No bears to show. Did you forget to install '
                               'the `coala-bears` package? Try `pip3 install '
                               'coala-bears`.')
         return
 
+    results = []
+    if args and args.json:
+        from coalib.output.JSONEncoder import create_json_encoder
+        JSONEncoder = create_json_encoder(use_relpath=args.relpath)
+
     for bear, sections in sorted(bears.items(),
                                  key=lambda bear_tuple:
                                  bear_tuple[0].name.lower()):
-        show_bear(bear,
-                  show_description,
-                  show_params,
-                  console_printer)
+        if args and (args.json or args.format):
+            results.append(bear)
+        else:
+            show_bear(bear,
+                      show_description,
+                      show_params,
+                      console_printer)
+
+    if args:
+        if args.json:
+            json_output = {'bears': results}
+            import json
+            if args.output:
+                filename = str(args.output[0])
+                with open(filename, 'w+') as fp:
+                    json.dump(json_output, fp,
+                              cls=JSONEncoder,
+                              sort_keys=True,
+                              indent=2,
+                              separators=(',', ': '))
+            else:
+                print(json.dumps(json_output,
+                                 cls=JSONEncoder,
+                                 sort_keys=True,
+                                 indent=2,
+                                 separators=(',', ': ')))
+        elif args.format:
+            print_bears_formatted(results)
 
```

Comment:
maybe put here
```
for bear in results:
    show_bear(...)
```
from above, to make this more readable in code. So you have the bear-gathering phase and the printing phase. However you need return statements then for each case from above, e.g.:
```
print_bears_formatted(...)
return
```

Reply:
Yeah makes sense. I'm kinda checking `args and (args.json or args.format)` condition for all the bears individually, which is unnecessary.

Reply:
I can't see scenario of error over here. So, by default coala.py returns 0 after calling show_bears function from ConsoleInteraction module. I haven't changed that part.

Reply:
> I can't see scenario of error over here. So, by default coala.py returns 0 after calling show_bears function from ConsoleInteraction module. I haven't changed that part.

What do you mean?

Reply:
https://github.com/coala/coala/pull/4726/commits/f039489890654bc7396f7a482c1c62fcc96bf571#diff-aed82cdd504caf613603b250c092f460L88 returns success - zero after calling `show_bears` function with appropriate entry point through `args` param. So, we don't need to return any error code from `show_bears` function.